packages/circuit-ui/components/ImageInput/ImageInput.docs.mdx

```diff
@@ -0,0 +1,45 @@
+import { Status, Props, Story } from '../../../../.storybook/components';
+
+# ImageInput
+
+<Status.Stable />
+
+The ImageInput component allows users to upload images. It can be used on its own or as part of a form.
+
+<Story id="forms-imageinput--base" />
+
+<Props />
+
+## Usage guidelines
+
+Refer to the `Stateful` component story for a simplified code usage example.
+
+- **Do** use the onChange callback to immediately trigger an image upload when the user has selected one, even when the component is being used as part of a form (it will make form submission faster).
+- **Do not** use the ImageInput component if the image is read-only for the user. Instead, use the visual component from the ImageInput's `component` prop by itself, for example the [Avatar](Components/Avatar).
+
+## Input states
+
+### Existing image
+
+When an `src` prop is provided, the ImageInput renders a "Clear" icon instead of the "Add" icon. It also forwards the prop to the visual component passed as a prop. Clicking the "Clear" icon clears the input.
+
+<Story id="forms-imageinput--with-image" />
+
+### Invalid
+
+When an image upload fails, set the `invalid` prop to true to render invalid styles, and show the user a clear and concise error message using the `validationHint` prop.
+
+<Story id="forms-imageinput--invalid" />
+
+## Custom component
+
+The ImageInput has drop-in support for the [Avatar](Components/Avatar) component as a visual element.
+
+However, where the end result image needs to be in a landscape view, a custom component with different dimensions can be used to reflect the final image's aspect ratio.
```

Comment:
That's quite a specific use case — there are other reasons why someone would want to use a custom component. Let's keep this generic.

Reply:
This was from the Figma docs, but I agree, will make it more generic 🙂